### PR TITLE
feat: use cost_data_type and cost_data_field for common spec (aws widget)

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [ ] Not that difficult
- [ ] Approved feature branch merge to master

### Description
Please review in order by commits!
This PR includes other bug fixes!

### CODE REVIEW

File: apps/web/src/services/dashboards/widgets/_helpers/__tests__/widget-inherit-options-helper.test.ts [hunk 1]
```diff
@@ -1,108 +1,268 @@
...
+            selection_type: 'MULTI',
+        },
+    },
+    order: ['region'],
+};
 
 describe('[Widget Inherit Options Helper] getInitialWidgetInheritOptions', () => {
-    it('test', () => {
-        expect(true).toBe(true);
+    it('should return empty object if there is no property in widget config even if stored inherit options exist and enabled', () => {
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_helpers/__tests__/widget-inherit-options-helper.test.ts:55]: praise: nice title

File: apps/web/src/services/dashboards/widgets/_configs/widget-options-schema.ts [hunk 2]
```diff
@@ -54,9 +55,9 @@ const WIDGET_OPTION_KEYS = {
     'filters.cost_additional_info_value': 'filters.cost_additional_info_value',
 } as const;
 
-type widgetOptionKey = keyof typeof WIDGET_OPTION_KEYS;
+type WidgetOptionKey = keyof typeof WIDGET_OPTION_KEYS;
```
Comment (skdud4659) [inline apps/web/src/services/dashboards/widgets/_configs/widget-options-schema.ts:58]: thx...

**Review by skdud4659 — APPROVED** [review]
LGTM!
thx for your effort🙏🏻!!

**Review by yuda110 — APPROVED** [review]
LGTM!!

**Pull request merged**